src/after-compile.ts

```diff
@@ -246,18 +246,20 @@ function provideDeclarationFilesToWebpack(
     }
 
     const outputFiles = getEmitOutput(instance, filePath);
-    const declarationFile = outputFiles
-      .filter(outputFile => outputFile.name.match(constants.dtsDtsxRegex))
-      .pop();
-    if (declarationFile !== undefined) {
-      const assetPath = path.relative(
-        compilation.compiler.context,
-        declarationFile.name
-      );
-      compilation.assets[assetPath] = {
-        source: () => declarationFile.text,
-        size: () => declarationFile.text.length
-      };
+    const declarationFiles: typescript.OutputFile[] = outputFiles
+      .filter((outputFile: typescript.OutputFile) => outputFile.name.match(constants.dtsDtsxRegex));
+
+    if (Array.isArray(declarationFiles)) {
```

Comment:
Yes, exactly! Due to the regex change, getEmitOutput will return both d.ts and d.ts.map when declarationMap is true. It felt logical to handle declaration & map files together since they are linked together inside the d.ts file, and besides the map file can't live without the declaration file. When it comes to performance loss for using a forEach instead of a pop() when it's only one should not a problem since we're not using a million and millions of files (i hope).